backend/api/auth.py

```
@@ -111,3 +142,125 @@ async def register(
+        else:
+            # Code validation failed - return appropriate error with proper status code
+            # Use 400 for client errors (invalid/expired codes)
+            raise HTTPException(
+                status_code=status.HTTP_400_BAD_REQUEST,
+                detail=result["error"]
+            )
+            
+    except HTTPException:
+        # Re-raise HTTP exceptions (don't catch our own exceptions)
+        raise
+    except Exception as e:
```

Ruff (0.13.2)

Do not catch blind exception: `Exception`

(BLE001)